Dockerfile

```diff
@@ -15,13 +15,27 @@ ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
 
 FROM common AS sgx
+ARG PSW_VERSION=2.13.103.1
+ARG DCAP_VERSION=1.10.103.1
 RUN apt update && \
-    apt install -y --no-install-recommends libsgx-dcap-ql libsgx-dcap-ql-dev libsgx-dcap-quote-verify libsgx-enclave-common libsgx-urts az-dcap-client && \
+    apt install -y --no-install-recommends \
+    az-dcap-client \
+    libsgx-dcap-ql=$DCAP_VERSION-bionic1 \
+    libsgx-dcap-ql-dev=$DCAP_VERSION-bionic1 \
+    libsgx-pce-logic=$DCAP_VERSION-bionic1 \
+    libsgx-qe3-logic=$DCAP_VERSION-bionic1 \
+    libsgx-dcap-quote-verify=$DCAP_VERSION-bionic1 \
+    libsgx-enclave-common=$PSW_VERSION-bionic1 \
+    libsgx-urts=$PSW_VERSION-bionic1 \
+    libsgx-enclave-common-dev=$PSW_VERSION-bionic1 \
+    libsgx-dcap-default-qpl=$DCAP_VERSION-bionic1 && \
     apt clean && apt autoclean
+# rename the softlink created by libsgx-dcap-default-qpl to avoid issues with az-dcap-client
+RUN mv /usr/lib/x86_64-linux-gnu/libdcap_quoteprov.so.1 /usr/lib/x86_64-linux-gnu/libdcap_quoteprov.so.1.intel
 
 FROM sgx AS sgx-dev
 RUN apt update && \
-    apt install -y libsgx-enclave-common-dev libsgx-ae-qve libsgx-ae-pce libsgx-ae-qe3 libsgx-qe3-logic libsgx-pce-logic && \
+    apt install -y libsgx-enclave-common-dev=$PSW_VERSION-bionic1 && \
```

Comment:
Does this work? In my understanding PSW_VERSION is out of scope (https://docs.docker.com/engine/reference/builder/#scope). If so, we should declare it here, but define its value before the first FROM (https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact).

Reply:
The build was successful when i tried it. But from the docs it looks like it should cause a problem. 
Just to be safe i will add the arg to the build step